src/ApiStatus.ts

```diff
@@ -0,0 +1,198 @@
+            this.currentStatus = this.parseApiStatus(apiStatus);
+            this.lastCheckTime = Date.now();            
+        }
+
+        return this.currentStatus;
+    }
+
+    private parseApiStatus(apiStatus: APIStatus): StatusEmbedState {
+        let cacheObject: { [key: string]: any } = {};
+
+        let onFire = false;
+        let allApisOK = true;
+        let allApisIssues = false;
+        let apiIssuesCounter = 0;
+        let apiCounter = 0;
+        let statusEmbed: StatusEmbedState = { api: {}, onFire: false, allApisOK: true, allApisIssues: false};
+
+        type RegionState = {"troubled": Array<string>, "up": Array<string>, "down": Array<string>};
+
+        for (let api in apiStatus) {
+            let regionStates: RegionState  = {"troubled": [], "up": [], "down": []};
+            let regionCounter = 0;
+            for (let region in apiStatus[api]) {
+                let regionState = apiStatus[api][region];
+                regionStates[regionState.state].push(region);
+                regionCounter++;
+            }
+
+            cacheObject[api] = {};
+            let retStr = "";
+            if (regionStates.troubled.length > 0) {
+                allApisOK = false;
+                retStr += ":warning:" + this.joinArray(regionStates.troubled) + "\n";
+            }
+            if (regionStates.down.length > 0) {
+                allApisOK = false;
+                retStr += ":x:" + this.joinArray(regionStates.down) + "\n";
+            }
+            if (regionStates.up.length > 0) {
+                allApisIssues = false;
+            }
+            apiIssuesCounter += regionStates.troubled.length + regionStates.down.length;
+            apiCounter += regionStates.troubled.length + regionStates.down.length + regionStates.up.length;
+
+            // API on fire, if all regions for one api have issues
+            if (regionStates.troubled.length + regionStates.down.length == regionCounter) {
+                onFire = true;
+            }
+
+            // only add api if it has issues
+            if (regionStates.troubled.length + regionStates.down.length > 0)
+                statusEmbed.api[api] = retStr;
+        }
+
+        statusEmbed["onFire"] = onFire || (apiIssuesCounter / apiCounter) > this.sharedSettings.apiStatus.apiOnFireThreshold;
+        statusEmbed["allApisOK"] = allApisOK;
+        statusEmbed["allApisIssues"] = allApisIssues;
+        return statusEmbed;
+    }
+
+    private joinArray(arr: Array<string>): string {
+        let retStr = ''
+        let i = 0;
+        for (let j = 0; j < arr.length; j++) {
+            let a = arr[j]
+            retStr += a + (j < arr.length - 1 ? ', ' : '')
+    
+            if (i % 4 == 3) {
+                retStr += '\n'
+            }
+            i++
+        }
+
+        // pad to minimum length
+        while(retStr.length < 'BR, EUNE, EUW, JP,'.length) {
+            retStr += " "
+        }
+        return retStr
```

Comment:
There are a bunch of missing semi-colons here. Also, `Array.prototype.join` is a thing.

Reply:
I don't think Array.prototype.join would help me much in this case. I only add a `\n` every 4 elements. This is not possible with `join` (unless I use `Array.slice` on 4 elements each, which would make the code less readable imo).